comp/core/tagger/taggerimpl/server/server.go

```diff
@@ -27,15 +28,30 @@ const (
 	streamKeepAliveInterval = 9 * time.Minute
 )
 
+// streamIDManager is used to generate unique ID's for incoming streaming requests
+// This unique ID is used to subscribe to the tagger
+// TODO: remove this struct when the protobuf of the stream request is updated to use filters.
+type streamIDManager int32
+
+func (s *streamIDManager) getNextUniqueID() string {
+	id := fmt.Sprintf("stream-client-%d", *s)
+	atomic.AddInt32((*int32)(s), 1)
+	return id
+}
```

Comment:
IMO this solution is fine. 
I'm wondering which other alternative you have considered. have you think about using a UUID generator?

Reply:
UUID can work too.

But I didn't spend so much time thinking about this because it is only meant to be a temporary solution until we update the protobuf definition. 

Ideally, I think the cluster check runner (or generally, the subscriber), should provide the subscription ID.
It is in that place that I plan to use UUID or something similar. 

On the server side, we can then append some prefix to the id that is set by the subscribing component.

As an example, a cluster check runner can subscribe to the remote cluster tagger by specifying a filter and a uuid (say `77ashyyqqq`). The cluster tagger server would register the id as `cluster-check-runner-streamer-77ashyyqqq`.